spec/services/idempotency_records/key_service_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe IdempotencyRecords::KeyService, type: :service do
+  subject(:result) { described_class.call(*key_parts) }
+
+  let(:key_parts) { [] }
+  let(:key_parts_reversed) { key_parts.reverse }
+
+  describe "#call" do
+    it "returns the same value when called twice" do
+      expect(result.idempotency_key).to eq(described_class.call(*key_parts).idempotency_key)
+    end
+
+    context "with one key_part" do
+      let(:key_parts) { ["key"] }
+
+      it "returns the same value when called twice" do
+        expect(result.idempotency_key).to eq(described_class.call(*key_parts).idempotency_key)
+      end
+    end
+
+    context "with multiple key_parts" do
+      let(:key_parts) { ["key1", "key2"] }
+
+      it "returns the same value when called twice" do
+        expect(result.idempotency_key).to eq(described_class.call(*key_parts).idempotency_key)
+      end
+
+      it "does not return the same value if the order of the key_parts changes" do
+        expect(result.idempotency_key).not_to eq(described_class.call(*key_parts_reversed).idempotency_key)
+      end
```

Comment:
I'm wondering if this is what we want. I fee like using `[payable.id, payable.amount]` should return the same key as `[payable.amount, payable.id]` 🤔

Reply:
Huh, interesting viewpoint. How would you implement that though? (just sorting? 🤔 )

Reply:
Yes, sorting the array before joining.

Reply:
Indeed I didn't think about boolean 🤔  Sorting is probably not possible then.

I was thinking about 2 services, doing similar things. If the args are in a different order it won't prevent duplication but it's probably very edge case. It's probably better to know that the order matters so we're careful.

Reply:
I think I'll just use a hash instead? then the order is no longer important and it's easy to think about and use.